packages/cdk8s/lib/app.ts

```diff
@@ -32,12 +37,140 @@ export class App extends Construct {
    * Synthesizes all manifests to the output directory
    */
   public synth(): void {
-    if (!fs.existsSync(this.outdir)) {
-      fs.mkdirSync(this.outdir);
+
+    fs.mkdirSync(this.outdir, { recursive: true });
+
+    // this is kind of sucky, eventually I would like the DependencyGraph
+    // to be able to answer this question.
+    const hasChartDependencies = App.prepare(this);
+
+    // Since we plan on removing the distributed synth mechanism, we no longer call `Node.synthesize`, but rather simply implement
+    // the necessary operations. We do however want to preserve the distributed validation.
+    App.validate(this);
+
+    this.produceManifests(hasChartDependencies);
+
+  }
+
+  /**
+   * Synthesize a single chart.
+   *
+   * Each element returned in the resulting array represents a different ApiObject
+   * that is contained in the chart.
+   *
+   * Note that the return value order is important. It is determined by the various dependencies between
+   * the constructs in the chart, where the first element is the one without dependencies, and so on...
+   *
+   * @returns An array of JSON objects.
+   * @param chart the chart to synthesize.
+   */
+  public static synthChart(chart: Chart): any[] {
```

Comment:
Let's make this internal:

```suggestion
  /**
   * Synthesize a single chart.
   *
   * Each element returned in the resulting array represents a different ApiObject
   * that is contained in the chart.
   *
   * Note that the return value order is important. It is determined by the various dependencies between
   * the constructs in the chart, where the first element is the one without dependencies, and so on...
   *
   * @returns An array of JSON objects.
   * @param chart the chart to synthesize.
   * @internal
   */
  public static _synthChart(chart: Chart): any[] {
```

Reply:
👍